### PR TITLE
Display changes in file versions tab view and detailsView

### PR DESCRIPTION
**Pull request by mjobst-necls**
## Description

I made some changes to improve the file version tab.
## Related Issue

Issue Tag: #26510
## Motivation and Context

This pull requests contains code to view the version file size on the file version tab.
Another little change was done for the favourite "star" button on the details view. The opacity of this button changes to 1 on mouse over by this pull request.
## How Has This Been Tested?

The test was ok for the following browsers:
- Internet Explorer 8 on Windows 7
- Google Chrome (current version) and Mozilla Firefox (current version) on Ubuntu 16.04
## Screenshots (if appropriate):

screenshot from 2016-10-29 21-51-33
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by mjobst-necls** [comment]
I already signed the CLA. Do I have to do an additional action?

File: apps/files_versions/css/versions.css [hunk 1]
```diff
@@ -12,40 +13,60 @@
...
+.versionsTabView .version-details {
+	text-align: left;
+}
+
+.versionsTabView .version-details > span {
+	padding: 0 10px;
+}
+
+.versionsTabView .version-details > .size {
+	vertical-align: super;
```
Comment (PVince81) [inline apps/files_versions/css/versions.css]: Not sure why "super" here ?
On Chromium for me the size appears under the date but not as superscript. It does look good already, so not sure why. If you wanted to shift it up a bit you could try `vertical-align: middle` and set a different `line-height`.

File: apps/files_versions/js/versionstabview.js [hunk 2]
```diff
@@ -10,14 +10,27 @@
 
 (function() {
 	var TEMPLATE_ITEM =
-		'<li data-revision="{{timestamp}}">' +
-		'<img class="preview" src="{{previewUrl}}"/>' +
-		'<a href="{{downloadUrl}}" class="downloadVersion"><img src="{{downloadIconUrl}}" />' +
-		'<span class="versiondate has-tooltip" title="{{formattedTimestamp}}">{{relativeTimestamp}}</span>' +
+                '<li data-revision="{{timestamp}}">' +
```
Comment (PVince81) [inline apps/files_versions/js/versionstabview.js]: Please use tabs instead of spaces in all the changed lines.

**Comment by davitol** [comment]
Using Chrome Version 54.0.2840.71 (64-bit) in OSX

PR applied:
screen shot 2016-11-02 at 10 49 38


PR not applied:
screen shot 2016-11-02 at 10 50 25

**Review by PVince81 — COMMENTED** [review]
Great PR, love it! Also works well.

There are a few minor issues, please see my review comments.

Would you mind adding the size field in the JS unit tests here https://github.com/owncloud/core/blob/v9.1.1/apps/files_versions/tests/js/versionstabviewSpec.js#L80 ?

**Comment by PVince81** [comment]
> Internet Explorer 8

By the way, since 9.2 OC only supports IE >= 11 so no need to test in much older versions :smile:

**Comment by davitol** [comment]
BTW also tested in Safari Version 9.1.2 (10601.7.7) and Firefox 49.0.2 and it went fine

**Comment by davitol** [comment]
> Another little change was done for the favourite "star" button on the details view. The opacity of this button changes to 1 on mouse over by this pull request.

It works but maybe ideally it would be great that the star will turn into yellow on mouse over as it happens in the files view. With this PR, the star keeps in grey. BTW it is a better behaviour than before. Great job!!

starlight

**Comment by mjobst-necls** [comment]
> Using Chrome Version 54.0.2840.71 (64-bit) in OSX

Can please someone tell me why chrome displays code different on OSX than on other OSs. Do you guys know a possibility how to test the code on OSX without apple hardware? I use a vm for tests on windows.

> It works but maybe ideally it would be great that the star will turn into yellow on mouse over as it happens in the files view. With this PR, the star keeps in grey. BTW it is a better behaviour than before. Great job!!

Thank you, I'm going to create another PR to cover your change requests regarding the color of the favorite star on hover and the JS unit test enhancements.

**Comment by PVince81** [comment]
@mjobst-necls a separate PR for the favorite star color is fine.

However could you add the JS unit test extension to this PR to make it complete since this enhancement is mostly about the versions tab ?

**Comment by mjobst-necls** [comment]
Ok, commit done.

File: apps/files_versions/js/versionstabview.js [hunk 3]
```diff
@@ -180,9 +193,13 @@
 
 		_formatItem: function(version) {
 			var timestamp = version.get('timestamp') * 1000;
+			var size = version.has('size') ? version.get('size') : 0;
 			return _.extend({
 				formattedTimestamp: OC.Util.formatDate(timestamp),
 				relativeTimestamp: OC.Util.relativeModifiedDate(timestamp),
+                                humanReadableSize: OC.Util.humanFileSize(size, true),
+                                altSize: n('files', '%n byte', '%n bytes', size),
+                                hasDetails: version.has('size'),
```
Comment (mjobst-necls) [inline apps/files_versions/js/versionstabview.js]: omg ^^. you're right ....

**Comment by PVince81** [comment]
Thanks. Apart from the missed tabs this looks good.

**Comment by PVince81** [comment]
:+1: merged, thanks a lot!

Added to features page: https://github.com/owncloud/core/wiki/ownCloud-9.2-Features

**Pull request merged**